pkgs/development/compilers/julia/default.nix

```diff
@@ -0,0 +1,136 @@
+{ stdenv
+, fetchurl
+# see https://github.com/JuliaLang/julia/blob/master/doc/build/build.md
+# not native
+, libunwind
+, pcre2
+, blas
+, lapack
+, mpfr
+, suitesparse
+, utf8proc
+, mbedtls
+, libssh2
+, curl
+, libgit2
+, patchelf
+, zlib
+, p7zip
+, enableMKL ? true # not free
+, libmFlavor ? "julia" # Julia's openlibm is the only libm we have packaged
+# native
+, m4
+, gfortran
+, perl
+, python3
+, which
+, cmake
+}:
+
+let
+  majorVersion = "1";
+  minorVersion = "5";
+  maintenanceVersion = "1";
+  version = "${majorVersion}.${minorVersion}.${maintenanceVersion}";
+in
+
+stdenv.mkDerivation rec {
+  pname = "julia";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://github.com/JuliaLang/julia/releases/download/v${version}/julia-${version}-full.tar.gz";
+    sha256 = "1m5iz5ikl0y1yvpmy7ys6a4l7xwn38mxsqm8g820gkf7rzyfn38x";
+  };
+
+  buildInputs = [
+    libunwind
+    pcre2
+    blas
+    lapack
+    mpfr
+    suitesparse
+    utf8proc
+    mbedtls
+    libssh2
+    curl
+    libgit2
+    patchelf
```

Comment:
`patchelf` should go into `nativeBuildInputs`. And why is it needed in a build from source?

Reply:
See https://github.com/JuliaLang/julia/blob/ff174120aad91e1a838b5cf99bd2b97c14aab2f5/contrib/fixup-libgfortran.sh#L34-L44 for example.